test/Rings/MPolyAnyMap/flattenings.jl

```diff
@@ -117,3 +115,198 @@ end
   @test Oscar.map_from_coefficient_ring_to_flattening(phi).(gens(R)) == phi.(Q.(S.(gens(R))))
 end
 
+@testset "flattenings of graded rings" begin
+  # towers of polynomial rings
+  R, (x, y) = polynomial_ring(QQ, [:x, :y], cached=false)
+  S, (u, v) = grade(polynomial_ring(R, [:u, :v], cached=false)[1])
+
```

Comment:
Does this creation of S also work with graded_polynomial_ring? 
If so, it might be appropriate to do this. (This is the case I expect)
If not, @fingolfin would certainly like to hear about the problem.